mpi4py.spec

```diff
@@ -386,13 +355,14 @@ mv build mpich
 # For SUSE:
 %{!?python3:%global python3 %{_bindir}/%{python_for_executables}}
 for py_site_arch in "${site_arches[@]}"; do
-    mkdir -p %{buildroot}/$py_site_arch/%{name}/tests
-    install -m 0755 test/test_io.py %{buildroot}/$py_site_arch/%{name}/tests/test_io_daos.py
-    sed -i -e '1i#!%{python3}' %{buildroot}/$py_site_arch/%{name}/tests/test_io_daos.py
+    mkdir -p %{buildroot}/%{_bindir}/
+    install -m 0755 test/test_io.py %{buildroot}/%{_bindir}/test_io_daos.py
```

Comment:
One concern here is if we need to install RPMs built with different python versions, this will make them incompatible.  Maybe we should install this script in the bin directory with the python version, e.g. `/usr/bin/test_io_daos_3.9.py` and then create a symlink to it w/o the python version, e.g. `ln -s /usr/bin/test_io_daos{_3.9,}.py`.

Reply:
The whole point here though is that currently we need to know the python version the binary was created with and that is what the ticket is asking to do away with.

How is the symlink created?  Who/what creates it?  When (if) there are multiple versions installed, how is it decided which one gets the symlink?

As far as having multiple python versioned packages installed, that's what the current packaging provides for, but it's not suitable per DAOS-15602.  I think we need to just decide that we will not allow multiple python versions for these packages.

Reply:
Looking at https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/9/html/installing_and_using_dynamic_programming_languages/assembly_packaging-python-3-rpms_installing-and-using-dynamic-programming-languages it would seem that using `%{_bindir}/test_io_daos.py` is correct, as long as we are including the python version in the package name - which we are.